src/commands/sauce.rs

```diff
@@ -121,15 +85,106 @@ pub async fn sauce(ctx: &Context, msg: &Message, args: Args) -> CommandResult {
 
 async fn call_user_out(ctx: &Context, msg: &Message) -> CommandResult {
     msg.author.dm(ctx, |m| {
-        m
-        .content("One of your message or the previous message must have an attachment **or** you must provide a valid URL as argument to the command")
-    }).await?;
+            m
+            .content("One of your message or the previous message must have an attachment **or** you must provide a valid URL as argument to the command")
+            }).await?;
     msg.react(ctx, ReactionType::Unicode("❌".into())).await?;
     Ok(())
 }
 
-fn get_first_attachment_url_from(msg: &Message) -> Option<Url> {
-    let attachment = msg.attachments.get(0)?;
+fn get_urls(msg: &Message, args: Args, prev_msg: Option<&Message>) -> Option<Vec<Url>> {
+    let mut res = if let Some(urls) = get_urls_from_message(msg, args) {
+        urls
+    } else {
+        Vec::new()
+    };
+
+    if let Some(mut urls) = msg
+        .referenced_message
+        .clone()
+        .and_then(|m| get_attachment_urls(&*m))
+    {
+        res.append(&mut urls);
+    }
+
+    if res.is_empty() {
```

Comment:
This won't get the sauce of an attachment if the user has an URL in their message, which is not what we want

Reply:
Misunderstood the logic, please add a comment somewhere in this function

Reply:
Added relevant comments